internal/controller/deployment_ to_pdb_controller.go

```diff
@@ -0,0 +1,164 @@
+package controllers
+
+import (
+	"context"
+
+	"k8s.io/api/apps/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/tools/record"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// DeploymentToPDBReconciler reconciles a Deployment object and ensures an associated PDB is created and deleted
+type DeploymentToPDBReconciler struct {
+	client.Client
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
+}
+
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;update;watch
+
+// Reconcile watches for Deployment changes (created, updated, deleted) and creates or deletes the associated PDB.
+// creates pdb with minAvailable to be same as replicas for any deployment
+func (r *DeploymentToPDBReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := log.FromContext(ctx)
+
+	// Fetch the Deployment instance
+	var deployment v1.Deployment
+	if err := r.Get(ctx, req.NamespacedName, &deployment); err != nil {
+		log.Error(err, "unable to fetch Deployment")
+		if client.IgnoreNotFound(err) == nil {
```

Comment:
nit:  if  errors.IsNotFound(err) { 
reads a little better Ignorenotfound is I am guesisnf mostly for the reutrn not ifs.

Reply:
And leave a comment about it being possible to leak for now since we're not using finalizers (or owner refences)